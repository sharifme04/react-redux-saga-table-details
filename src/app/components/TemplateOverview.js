import React from "react";
import PropTypes from "prop-types";
import OverviewRow from "./OverviewRow";

const TemplateOverview = ({ isLoading, templates }) => {
  return (
    <div className="container">
      <h2>Template Overview</h2>
      <hr />
      {isLoading ? (
        <div className="d-flex justify-content-center">
          <div className="spinner-border text-info spin-size" role="status">
            <span className="sr-only">Loading...</span>
          </div>
        </div>
      ) : templates?._data?.length ? (
        <table className="table table-hover">
          <thead>
            <tr>
              <th>Template ID</th>
              <th>Name</th>
              <th>Version</th>
              <th>Valid From</th>
              <th>Valid To</th>
              <th>Template Details</th>
              <th>Template Delete</th>
            </tr>
          </thead>
          <tbody>
            {templates?._data?.map((template) => (
              <OverviewRow key={template.templateId} template={template} />
            ))}
          </tbody>
        </table>
      ) : (
        <div className="text-center"> ohh!!! no template available !!</div>
      )}
    </div>
  );
};

TemplateOverview.propTypes = {
  isLoading: PropTypes.bool,
  templates: PropTypes.shape({
    status: PropTypes.number,
    message: PropTypes.string,
    _data: PropTypes.array,
  }),
};

export default TemplateOverview;
