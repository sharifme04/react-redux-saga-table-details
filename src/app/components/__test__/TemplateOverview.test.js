import React from "react";
import { shallow } from "enzyme";
import TemplateOverview from "../TemplateOverview";

const templates = {
  message: "get_templates_success",
  status: 0,
  _data: [
    {
      conditions: null,
      name: "basic_user",
      optionalDocuments: {
        assetType: {
          conditions: { topLevelDomain: ".edu" },
          values: { emailAddress: true },
        },
      },
      partnerId: "59c25c00-78cd-11ea-ac9b-bf680aaa6d4b",
      requiredDocuments: {
        assetType: {
          conditions: { topLevelDomain: ".edu" },
          values: { emailAddress: true },
        },
      },
      templateId: "487c5b70-9516-11ea-bbfa-ed3b490e43f0",
      type: "basic",
      validFrom: "04-05-2020",
      validTo: "21-05-2020",
      version: 2,
    },
  ],
};

describe("<TemplateOverview />", () => {
  test("renders initial", () => {
    const component = shallow(<TemplateOverview templates={templates} />);
    expect(component).toMatchSnapshot();
  });

  test("return the default array when there is no data to map through", () => {
    const component = shallow(<TemplateOverview />);
    expect(component).toMatchSnapshot();
  });

  test("does not break without value", () => {
    const component = shallow(<TemplateOverview />);
    expect(component.find("h2")).toHaveLength(1);
    expect(component.find("table")).toHaveLength(0);
  });

  test("does not break with an empty array", () => {
    const component = shallow(<TemplateOverview fakeTemplates={[]} />);
    expect(component.find("table")).toHaveLength(0);
  });
});
