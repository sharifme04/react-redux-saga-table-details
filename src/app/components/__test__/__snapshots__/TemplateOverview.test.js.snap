// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TemplateOverview /> renders initial 1`] = `
<div
  className="container"
>
  <h2>
    Template Overview
  </h2>
  <hr />
  <table
    className="table table-hover"
  >
    <thead>
      <tr>
        <th>
          Template ID
        </th>
        <th>
          Name
        </th>
        <th>
          Version
        </th>
        <th>
          Valid From
        </th>
        <th>
          Valid To
        </th>
        <th>
          Template Details
        </th>
        <th>
          Template Delete
        </th>
      </tr>
    </thead>
    <tbody>
      <OverviewRow
        key="487c5b70-9516-11ea-bbfa-ed3b490e43f0"
        template={
          Object {
            "conditions": null,
            "name": "basic_user",
            "optionalDocuments": Object {
              "assetType": Object {
                "conditions": Object {
                  "topLevelDomain": ".edu",
                },
                "values": Object {
                  "emailAddress": true,
                },
              },
            },
            "partnerId": "59c25c00-78cd-11ea-ac9b-bf680aaa6d4b",
            "requiredDocuments": Object {
              "assetType": Object {
                "conditions": Object {
                  "topLevelDomain": ".edu",
                },
                "values": Object {
                  "emailAddress": true,
                },
              },
            },
            "templateId": "487c5b70-9516-11ea-bbfa-ed3b490e43f0",
            "type": "basic",
            "validFrom": "04-05-2020",
            "validTo": "21-05-2020",
            "version": 2,
          }
        }
      />
    </tbody>
  </table>
</div>
`;

exports[`<TemplateOverview /> return the default array when there is no data to map through 1`] = `
<div
  className="container"
>
  <h2>
    Template Overview
  </h2>
  <hr />
  <div
    className="text-center"
  >
     ohh!!! no template available !!
  </div>
</div>
`;
